src/components/TableFilterInline.js

```diff
@@ -0,0 +1,266 @@
+import ListItemText from '@material-ui/core/ListItemText';
+import MenuItem from '@material-ui/core/MenuItem';
+import Select from '@material-ui/core/Select';
+import { withStyles } from '@material-ui/core/styles';
+import PropTypes from 'prop-types';
+import React from 'react';
+import TableHeadRow from './TableHeadRow';
+
+
+export const defaultFilterInlineStyles = theme => ({
+  root: {
+    backgroundColor: theme.palette.background.default,
+    padding: '24px 24px 36px 24px',
+    fontFamily: 'Roboto',
+  },
+  input: {
+    fontSize: '0.8125rem',
+    fontWeight: '400',
+  },
+  noMargin: {
+    marginLeft: '0px',
+  },
+  reset: {
+    alignSelf: 'left',
+  },
+  resetLink: {
+    marginLeft: '16px',
+    fontSize: '12px',
+    cursor: 'pointer',
+  },
+  filtersSelected: {
+    alignSelf: 'right',
+  },
+  /* checkbox */
+  checkboxListTitle: {
+    marginLeft: '7px',
+    marginBottom: '8px',
+    fontSize: '14px',
+    color: theme.palette.text.secondary,
+    textAlign: 'left',
+    fontWeight: 500,
+  },
+  checkboxFormGroup: {
+    marginTop: '8px',
+  },
+  checkboxFormControl: {
+    margin: '0px',
+  },
+  checkboxFormControlLabel: {
+    fontSize: '15px',
+    marginLeft: '8px',
+    color: theme.palette.text.primary,
+  },
+  checkboxIcon: {
+    width: '32px',
+    height: '32px',
+    padding: '0px',
+  },
+  checkbox: {
+    '&$checked': {
+      color: theme.palette.primary.main,
+    },
+  },
+  checked: {},
+});
+
+class TableFilterInline extends React.Component {
+  static propTypes = {
+    /** Data used to populate filter dropdown/checkbox */
+    filterData: PropTypes.array.isRequired,
+    /** Data selected to be filtered against dropdown/checkbox */
+    filterList: PropTypes.array.isRequired,
+    /** Options used to describe table */
+    options: PropTypes.object.isRequired,
+    /** Callback to trigger filter update */
+    onFilterUpdate: PropTypes.func,
+    /** Callback to trigger filter reset */
+    onFilterRest: PropTypes.func,
+    /** Extend the style applied to components */
+    classes: PropTypes.object,
+  };
+
+  handleCheckboxChange = (index, value, column) => {
+    this.props.onFilterUpdate(index, value, column, 'checkbox');
+  };
+
+  handleDropdownChange = (event, index, column) => {
+    const labelFilterAll = this.props.options.textLabels.filter.all;
+    const value = event.target.value === labelFilterAll ? [] : [event.target.value];
+    this.props.onFilterUpdate(index, value, column, 'dropdown');
+  };
+
+  handleMultiselectChange = (index, value, column) => {
+    this.props.onFilterUpdate(index, value, column, 'multiselect');
+  };
+
+  handleTextFieldChange = (event, index, column) => {
+    this.props.onFilterUpdate(index, event.target.value, column, 'textField');
+  };
+
+  handleCustomChange = (value, index, column) => {
+    this.props.onFilterUpdate(index, value, column.name, column.filterType);
+  };
+
+  // TODO extend TableFilter
+  renderCheckbox(column, index) {
```

Comment:
I'd like to see each of these have the `filterList` icon (https://material-ui.com/components/material-icons/) as an input adornment (https://material-ui.com/components/text-fields/#input-adornments) prefix.

Reply:
As below? For checkboxes I skipped it.

<img width="1003" alt="Screenshot 2020-01-15 at 11 41 13" src="https://user-images.githubusercontent.com/849975/72427449-0a8a3400-378c-11ea-9276-6b8aed99b1ff.png">